dart/lib/src/sentry_span_interface.dart

```diff
@@ -0,0 +1,33 @@
+import '../sentry.dart';
+
+abstract class ISentrySpan {
+  ISentrySpan startChild(
+    String operation, {
+    String? description,
+  });
+
+  void setTag(String key, String value);
+
+  void removeTag(String key);
+  // exposed to Transaction too but it does not get sent to the backend since transactions dont have the data bag
+  void setData(String key, dynamic value);
+// exposed to Transaction too but it does not get sent to the backend since transactions dont have the data bag
+  void removeData(String key);
+
+  // Future because finish calls hub.captureTransaction, should we do a Future or not?
+  Future<void> finish({
+    SpanStatus? status,
+  });
```

Comment:
makes sense to keep it as a Future since the user can decide whether to wait or not otherwise it could lead to back pressure problems